src/mongoc/mongoc-client.c

```diff
@@ -672,9 +662,48 @@ _mongoc_client_recv_gle (mongoc_client_t  *client,
  *
  *--------------------------------------------------------------------------
  */
+mongoc_client_t *
+mongoc_client_new(const char *uri_string)
+{
+   mongoc_uri_t *uri;
+   mongoc_topology_t *topology;
+
+   if (!uri_string) {
+      uri_string = "mongodb://127.0.0.1/";
+   }
+
+   if (!(uri = mongoc_uri_new(uri_string))) {
+      return NULL;
+   }
+
+   topology = mongoc_topology_new(uri);
+
+   mongoc_uri_destroy (uri);
+
+   return _mongoc_client_new(uri_string, topology);
+}
```

Comment:
I'm very confused by this function.
It seems to be doing exactly the same as _mongoc_client_new() - defaulting to mongodb://127.0.0.1 if no uri_string is given. And validate the uri_string.
I don't quite understand the need for this wrapper?